app/src/main/java/com/nextcloud/talk/models/json/chat/ChatUtils.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Nextcloud Talk application
+ *
+ * @author Mario Danic
+ * Copyright (C) 2017-2018 Mario Danic <mario@lovelyhq.com>
+ * @author Marcel Hibbe
+ * Copyright (C) 2021 Marcel Hibbe <dev@mhibbe.de>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.nextcloud.talk.models.json.chat
+
+import java.util.*
+
+class ChatUtils {
+    companion object {
+        fun getParsedMessage(message: String?, messageParameters: HashMap<String?, HashMap<String?, String?>>?):
+                String? {
+            var message = message
+            if (messageParameters != null && messageParameters.size > 0) {
+                for (key in messageParameters.keys) {
+                    val individualHashMap = messageParameters[key]
+                    val type = individualHashMap?.get("type")
+                    if (type == "user" || type == "guest" || type == "call") {
```

Comment:
type can be null if individualHashMap is null.
When you use "user" == type then it will not crash with NPE.

Reply:
this will not throw a NPE. 
This would resolve to  `null == "user"` which is "false"
I guess you were thinking about java where it's better to check for
`"mystring".equals(value)`
in case of
`value.equals("mysting")` (-> NPE if value is null)
?

Reply:
Right, bad Java habits ;-)